std/container/rbtree.d

```diff
@@ -801,9 +800,7 @@ if (is(typeof(binaryFun!less(T.init, T.init))))
 
     static private Node allocate(Elem v)
     {
-        auto result = allocate();
```

Comment:
The idea actually comes from the original library, [where I had allocators](https://github.com/schveiguy/dcollections/blob/master/dcollections/RBTree.d#L1244).

Since this is inside the abstraction for allocation, it is OK, as once std.experimental.allocators is ready, this will get noticed.

Thanks for reviewing, I just logged in this morning to do so and approve, but you beat me to it.